File: 2026/04-April/17-Friday/solution.java
// Minimum Absolute Distance Between Mirror Pairs

import java.util.*;

class Solution {
    public int minMirrorPairDistance(int[] nums) { 
        Map<Integer, Integer> map = new HashMap<>();
        int ans = Integer.MAX_VALUE;

        for (int j = 0; j < nums.length; j++) {
            int num = nums[j];

            if (map.containsKey(num)) {
                ans = Math.min(ans, j - map.get(num));
            }

            int rev = reverse(num);
            map.put(rev, j);
        }

        return ans == Integer.MAX_VALUE ? -1 : ans;
    }

    private int reverse(int x) {
        int rev = 0;
        while (x > 0) {
            rev = rev * 10 + x % 10;
            x /= 10;
        }
        return rev;
    }
}

File: 2026/04-April/17-Friday/solution.kt
// Minimum Absolute Distance Between Mirror Pairs

class Solution {
    fun minMirrorPairDistance(nums: IntArray): Int {
        val map = HashMap<Int, Int>()
        var ans = Int.MAX_VALUE

        for (j in nums.indices) {
            val num = nums[j]
 
            if (map.containsKey(num)) {
                ans = minOf(ans, j - map[num]!!)
            }

            val rev = reverse(num)
            map[rev] = j
        }

        return if (ans == Int.MAX_VALUE) -1 else ans
    }

    private fun reverse(x: Int): Int {
        var num = x
        var rev = 0
        while (num > 0) {
            rev = rev * 10 + num % 10
            num /= 10
        }
        return rev
    }
}
